lib/src/common/disposable.dart

```diff
@@ -126,10 +126,14 @@ typedef Future<dynamic> Disposer();
 /// Allows the creation of managed objects, including helpers for common
 /// patterns.
 ///
-/// There are four ways to consume this class: as a mixin, a base class,
-/// an interface, and a concrete class used as a proxy. All should work
-/// fine but the first is the simplest and most powerful. Using the class
-/// as an interface will require significant effort.
+/// We recommend consuming this class in one of two ways.
+///
+///   1. As a base class
+///   2. As a concrete proxy
+///
+/// We do not recommend using this class as a mixin or as an interface.
+/// Use as a mixin can cause the `onDispose` method to be shadowed. Use
+/// as an interface is just cumbersome.
```

Comment:
It might be valuable to say something about, `if an interface is required use DispoablabeManagerV777` here.